solr/core/src/java/org/apache/solr/request/macro/MacroExpander.java

```diff
@@ -136,14 +136,15 @@ private String _expand(String val) {
       }
       else if (idx < 0) {
         if (sb == null) return val;
-        sb.append(val.substring(start));
+        sb.append(val, start, val.length());
         return sb.toString();
       }
 
       // found unescaped "${"
+      final int matchedStart = idx;
       idx += macroStart.length();
```

Comment:
Can now remove idx manipulation until later when idx & start are changed in one line
```suggestion
```